### PR TITLE
chore(vue): update to vue 2.7

### PR DESCRIPTION
**Pull request by mdarrik**
- feat: replace vue-composition-api with vue 2.7
- chore: update to vite v3 & vue-2.7 vite plugin
- chore: update to use latest @nuxtjs-composition-api
- docs: adds changeset

## Why are these changes introduced?

Addresses [ENG-8369](https://nacelle.atlassian.net/browse/ENG-8369) -  make `@nacelle/vue` compatible with `@nuxtjs/composition-api` 0.33+ 

## What is this pull request doing?

1. Updates `@nacelle/vue` to replace `@vue/composition-api` with vue 2.7 & updates tests accordingly (vue 2.7 unwraps refs on, the vm object, which was not happening with `@vue/composition-api`, breaking a handful of tests).
2. Updates `@nacelle/vue` to vite 3 to use the `@vitejs/plugin-vue2` which is recommended for vite + vue 2.7.
3. Updates `@nacelle/vue` to use `jest@29`
4. Updates `examples/nuxt-2-composition-api` to use `@nuxtjs/composition-api@^0.33` & updates the underlying Nuxt version to use vue 2.7.

## How to Test

1. Checkout this branch `mdarrik/ENG-8369-update-nacelle-vue-to-vue-2-7`
2. From the root of the repo run `npm run bootstrap` 
3. Navigate to `packages/vue`
4. Run `npm run test` all tests should pass.
5. Run `npm run build` build should succeed
6. Run `npm link`
7. Navigate to `examples/nuxt-2-composition-api`
8. Make sure your `.env` file is setup with a Nacelle v1 space. 
9. Run `npm link @nacelle/vue` 
10. Run `npm run generate` - should generate your pages successfully. 
11. Run `npx serve dist/` - should be able to navigate to products/collections pages that render correctly

## Checklist

- [x] This Pull Request aligns with `nacelle-js`' [Code of Conduct](https://github.com/getnacelle/nacelle-js/blob/main/CODE_OF_CONDUCT.md#code-of-conduct).
- [x] You can follow your own "How to Test" instructions and get the expected result.
- [x] Screenshots, videos, etc. in the PR description are free of brand names and sensitive details.
- [x] Created a [changeset](https://github.com/changesets/changesets) (if the change should appear in a changelog).
- [x] Updated the `README.md` with documentation changes (if applicable).

[ENG-8369]: https://nacelle.atlassian.net/browse/ENG-8369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/healthy-mayflies-drop.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'nuxt-2-composition-api': minor
+'@nacelle/vue': minor
+---
+
+Replaces `@vue/composition-api` with Vue 2.7
```
Comment (NWRichmond) [inline .changeset/healthy-mayflies-drop.md:6]: Might be worth adding some details here about compatibility with `@nuxtjs/composition-api` versions. For example, that this version breaks compatibility with `@nuxtjs/composition-api` versions `0.32.0` and below, and that `@nacelle/vue` is now only expected to be compatible with the `0.33.x` versions of `@nuxtjs/composition-api` that are backed by Vue 2.7.

Comment (NWRichmond) [reply .changeset/healthy-mayflies-drop.md:6]: Might also be worth noting that `@nacelle/vue` also now requires transpilation config as described in the README.

**Review by NWRichmond — APPROVED** [review]
Awesome! The `@nacelle/vue` tests are passing, and when using a version of `examples/nuxt-2-composition-api` modified to use `@nacelle/compatibility-connector`, the PDP and collection PLPs function as expected:

https://user-images.githubusercontent.com/5732000/216214395-bc5ff940-a705-4fab-b291-1e93f6e8b2ca.mov

The only thing I'd recommend before merging is adding some more details to the changeset.

Really nice work with this PR! Thank you, @mdarrik!

**Review by KrisQ — APPROVED** [review]
Looks good thank you!

File: .changeset/healthy-mayflies-drop.md (hunk 1 above)
Comment (mdarrik) [reply .changeset/healthy-mayflies-drop.md]: @NWRichmond would you mind giving the updates I made a once over?

**Review by NWRichmond — APPROVED** [review]
Sweet! Thanks for those updates, @mdarrik.

**Review by KrisQ — APPROVED** [review]
Still looks good!

**Pull request merged**